x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/cleaner/CleanerService.java

```diff
@@ -242,14 +243,16 @@ public void cancel() {
     static class DefaultExecutionScheduler implements ExecutionScheduler {
 
         @Override
-        public TimeValue nextExecutionDelay(DateTime now) {
+        public TimeValue nextExecutionDelay(ZonedDateTime now) {
             // Runs at 01:00 AM today or the next day if it's too late
-            DateTime next = now.withTimeAtStartOfDay().plusHours(1);
+            ZonedDateTime next = now.toLocalDate()
+                .atStartOfDay(now.getZone())
+                .plusHours(1);
             // if it's not after now, then it needs to be the next day!
             if (next.isAfter(now) == false) {
                 next = next.plusDays(1);
             }
-            return TimeValue.timeValueMillis(next.getMillis() - now.getMillis());
+            return TimeValue.timeValueMillis(next.toInstant().toEpochMilli() - now.toInstant().toEpochMilli());
```

Comment:
minor nit: you could simplify this to `Duration.between(now, next).toMillis()` (pay attention that you have to switch the order in that case)